ibis/pandas/context.py

```diff
@@ -0,0 +1,97 @@
+import abc
+import operator
+
+import six
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Context(object):
```

Comment:
Yeah this isn't named very well.

The idea this encapsulates is really something like `AggregationContext`.

For any particular aggregation such as `sum`, `mean`, etc we need to decide based on the presence or absence of other expressions like `group_by` and `order_by` whether we should call `groupby(...).transform`, `groupby(...).apply`, `groupby(...).expanding().<method>`, `groupby(...).rolling().<method>` or just `groupby(...).<method>` on the data.